test/TestOverrides.cpp

```diff
@@ -0,0 +1,85 @@
+#include <jni.h>
+#include "API.h"
+#include "Test.h"
+#include <string.h>
+#include <assert.h>
+
+static jobject s_CustomClassLoader = 0;
+static jmethodID s_ClassForNameMethod = 0;
+static int s_ClassLoaderCallCount = 0;
+
+struct JNIOverrides
+{
+    typedef jclass(*FindClass)(JNIEnv*, const char*);
+
+    FindClass findClass;
+};
+
+jclass FindClass(JNIEnv* env, const char* name)
+{
+    // Taken from Unity code
+    env->ExceptionClear();
+    jclass clazz = env->FindClass("java/lang/Class");
+    jboolean initialize = JNI_TRUE;
+    jstring java_name = env->NewStringUTF(name);
+    jclass klass = (jclass)env->CallStaticObjectMethod(clazz, s_ClassForNameMethod, java_name, initialize, s_CustomClassLoader);
+    env->DeleteLocalRef(java_name);
+    env->DeleteLocalRef(clazz);
+    s_ClassLoaderCallCount++;
+    return klass;
+}
+
+void AllocateClassLoader(JNIEnv* env)
+{
+    jclass jniBridgeClass = env->FindClass("bitter/jnibridge/JNIBridge");
+    jclass classClass = env->GetObjectClass(jniBridgeClass);
+
+    // This piece of code taken from Unity (EntryPoint.cpp void InitJni(JavaVM* jvm, jobject obj, jobject context))
+    jclass classLoaderClass = env->FindClass("java/lang/ClassLoader");
+    jmethodID getClassLoaderMethod = env->GetMethodID(classClass, "getClassLoader", "()Ljava/lang/ClassLoader;");
+    jobject temp = env->CallObjectMethod(jniBridgeClass, getClassLoaderMethod);
+    // Important, need to do ExceptionCheck before NewGlobalRef, otherwise JNI throws warning
+    if (env->ExceptionCheck())
+        return;

```

Comment:
Does look right. You should only need exception check. This should only be the case when exception is actually thrown. And if it is thrown, you should do something about it. Now you just ignore it and postpone the failure to the next JNI call.

Reply:
Fixed